<launch>

    <arg name="duckiebot" default="duckiebot" />

    <arg name="tag_id" default="20" doc="ID of tracked tag" />

    <arg name="cam_frame" default="camera" doc="tf frame id for camera frame" />
    <arg name="vehicle_frame" default="car" doc="tf frame id for vehicle frame"/>
    <arg name="world_frame" default="world" doc="tf frame id for world frame"/>

    <arg name="is_bag" default="false" doc="program executed on bag (sim_time !)"/>

    <!-- Set sim_time if required (bag) -->
    <rosparam param="use_sim_time" subst_value="True">$(arg is_bag)</rosparam>

    <!-- Tf static transformations -->
    <include file="$(find duckietown-intnav)/launch/tf_tree.launch"> 
        <arg name="duckiebot" value="$(arg duckiebot)" />
        <arg name="cam_frame" value="$(arg cam_frame)" />
        <arg name="vehicle_frame" value="$(arg vehicle_frame)" />
        <arg name="world_frame" value="$(arg world_frame)" />
    </include>

    <!-- Image preprocessing -->
    <include file="$(find duckietown-intnav)/launch/image_preprocessing.launch"> 
        <arg name="duckiebot" value="$(arg duckiebot)" />
    </include>

    <!-- Intersection Map visualization -->
    <include file="$(find duckietown-intnav)/launch/imap_server.launch"> 
        <arg name="duckiebot" value="$(arg duckiebot)" />
    </include>

    <!-- April tag localization -->
    <include file="$(find duckietown-intnav)/launch/april_localization.launch"> 
        <arg name="duckiebot" value="$(arg duckiebot)" />
        <arg name="tag_id" value="$(arg tag_id)" />
        <arg name="cam_frame" value="$(arg cam_frame)" />
        <arg name="vehicle_frame" value="$(arg vehicle_frame)" />
        <arg name="world_frame" value="$(arg world_frame)" />       
    </include>

</launch>